### DISCUSSION TITLE
Adding MINIO_SERVER_URL breaks the console login

### DISCUSSION THREAD

**Discussion by lcfd** [minio/minio#15849] [Q&A] [answered]
## Expected Behavior
I expect to be able to login into the Minio Console and have the sharing links working. 
Minio should work behind a domain and TLS.

## Current Behavior

After entering the credentials, the loading spinner starts and ends after a long time.
The result is an error at the top of the page.

`Post "https://minio-api.example.com/": dial tcp xxx.xxx.xxx.xxx:443: i/o timeout`

## Steps to Reproduce (for bugs)

1. Add MINIO_SERVER_URL environment variable

## Context

I'm doing the deployment using Docker and Traefik.

## Regression
Is this issue a regression? No, we have been trying for a few months but we have always had the same problem.

## Your Environment

* Version used (`minio --version`): minio/minio:RELEASE.2022-10-08T20-11-00Z
* Server setup and configuration: Docker and Traefik
* Operating System and version (`uname -a`): Ubuntu 20.04.5 LTS (GNU/Linux 5.4.0-125-generic x86_64)

**Comment by ravindk89** [comment]
There are a few things to check here:

- Does the TLS certificate you passed cover that domain properly?
- Does that URL resolve properly from the host?
- Do you have firewall rules in place blocking access to privileged ports?
- How is your Traefik proxy configured


Lets start with those items and work down from there.

**Reply by lcfd** [reply]
Hi! Thank you for the reply.
I hope I can give more context via this answer.

* Does the TLS certificate you passed cover that domain properly?
	* The console and the API load correctly behind the TLS certificate. The only difference when using `MINIO_SERVER_URL` is that I can’t do login.
	* Do I need to configure something about TLS also in the volume of the container or certain Minio settings?
* Does that URL resolve properly from the host?
	* Yes, both the console and the API responds correctly by using the configured domain.
	* When I do the login I get this error
    ```
    code: 401
    detailedMessage: "Post \"https://minio-api.example.com/\": dial tcp xxx.xxx.xxx.xxx:443: i/o timeout"
    message: "invalid Login"
    ```
* Do you have firewall rules in place blocking access to privileged ports?
	* The port used in the error should be available (443) because is the port that we are using for HTTPS.
* How is your Traefik proxy configured
```yml
labels:
      - "traefik.enable=true"
      - "traefik.http.routers.miniobrowserconsole.service=miniobrowserconsole"
      - "traefik.http.routers.miniobrowserconsole.rule=Host(`minio.example.com`)"
      - "traefik.http.routers.miniobrowserconsole.entrypoints=https"
      - "traefik.http.routers.miniobrowserconsole.tls=true"
      - "traefik.http.services.miniobrowserconsole.loadbalancer.server.port=9001"
      - "traefik.http.services.miniobrowserconsole.loadbalancer.passhostheader=true"
      - "traefik.http.routers.minioapi.service=minioapi"
      - "traefik.http.routers.minioapi.rule=Host(`minio-api.example.com`)"
      - "traefik.http.routers.minioapi.entrypoints=https"
      - "traefik.http.routers.minioapi.tls=true"
      - "traefik.http.services.minioapi.loadbalancer.server.port=9000"
      - "traefik.http.services.minioapi.loadbalancer.passhostheader=true"
```

**Reply by ravindk89** [reply]
If you use `mc` to access the deployment using the same domain, do you get a similar error? `mc` might give more debugging information as well.

You can also run `mc admin trace` against the MinIO Server and see what happens when the login fails - you might get more information there, both with and without the `--debug` flags.

**Reply by lcfd** [reply]
Thank you for the suggestion! I haven't tried with `mc`.
I hope to find some more details using this tool.

Anyway, if there is a **good** guide that teaches how to put Mino into production by using Docker (compose) and Traefik, I'm all ears.

**Reply by ravindk89** [reply]
Re-writing our compose tutorial is on our list, but it wont be targeted for production - our opinion is that for production containerized environments, go with Kubernetes and use our Kubernetes Operator.

So when we do have the cycles to work on it, the Compose tutorial will be more focused around local development, and likely will not involve setting up proxies or load balancers (unless necessary to get that particular stack to work)

with regards to proxies/load balancers in general, that is something we are still discussing. I think we'll need to publish integration guides for NGinx and maybe HAProxy - not sure about traefik yet, though it is popular. But those are likely going to be standalone guides *or* targeted to Kubernetes, where we most often see difficulty in configuring those services.

Appreciate your understanding there.

**Reply by ludelafo** [reply]
Hey @lcfd, I'm having the same issue as you. Did you find a solution?

When I try with `mc`, I'm having the following output:

```sh
❯ export MC_HOST_myalias=https://<username>:<password>@<host>
❯ mc ls myalias
mc: <ERROR> Unable to list folder. Get "https://<host>/": dial tcp: lookup <host>: i/o timeout
```

I'm able to access my files using [CyberDuck](https://cyberduck.io/) though.

**Reply by 5hanth** [reply]
facing same issue with [coolify](https://github.com/coollabsio/coolify) which also uses traefik. coolify assigns one hostname route per service created. So directing traffic to two different ports seems to break the console login UI.

**Comment by rdelcorro** [comment, marked as answer]
If anyone is having this issue, I found the solution. Basically you need two sub domains or routers in traefik. One for the api and another for the console

You want to see something like this on startup:

```
Version: RELEASE.2024-01-18T22-51-28Z (go1.21.6 linux/amd64)
Status:         1 Online, 0 Offline. 
S3-API: https://minioapi.mydomain.com 
Console: https://minio.mydomain.com 
```

You do that by setting this two env vars
```
MINIO_SERVER_URL=https://minioapi.mydomaincom
MINIO_BROWSER_REDIRECT_URL=https://minio.mydomain.com
```

In traefik ( I use docker labels but any config works):

```
    labels:
      - "traefik.http.routers.minio-rtr.service=minio"
      - "traefik.http.routers.minio2-rtr.service=minio2"
      - "traefik.http.routers.minio2-rtr.entrypoints=https"
      - "traefik.http.routers.minio2-rtr.tls=true"
      - "traefik.http.services.minio2.loadbalancer.server.port=9000"
      - "traefik.http.routers.minio2-rtr.rule=Host(`minioapi.mydomain.com`)"

      - "traefik.http.routers.minio-rtr.entrypoints=https"
      - "traefik.http.routers.minio-rtr.rule=Host(`minio.mydomain.com`)"
      - "traefik.http.routers.minio-rtr.tls=true"
      - "traefik.http.services.minio.loadbalancer.server.port=80"

```

I am starting the server with this args:
`server /data --console-address ":80"`

so the console listens on port 80 and the api in the default 9000
